analysis_pipeline.py

```diff
@@ -0,0 +1,279 @@
+from datetime import timedelta
+import pandas as pd
+import numpy as np
+import pvlib
+import rdtools
+import logging
+from timezonefinder import TimezoneFinder
+
+import logger
+logger.init()
+    
+def get_timezone(latitude, longitude):
+    """
+    returns: timezone (str)
+    """
+    tf = TimezoneFinder()
+    tz = tf.timezone_at(lat=latitude, lng=longitude)
+    logging.info("timezone for location {}, {} is {}".format(latitude, longitude, tz))
+    
+    return tz
+
+def join(dataframe, series, name):
+    "Helper function to rename series and join to dataframe"
+    series = series.rename(name)
+    dataframe = dataframe.join(series)
+    return dataframe
+
+def rename_columns(df):
+    """
+    Best guess at renaming columns. Ignores units.
+    """
+    try:
+        df.columns = [col.decode('utf-8') for col in df.columns]
+    except AttributeError:
+        pass  # Python 3 strings are already unicode literals
+
+    columns = [name.lower() for name in df.columns]
+
+    rename_dictionary = {}
+    for name in columns:
+        if 'power' in name: 
+            rename_dictionary[name] = 'power'
+        elif 'temperature' in name or 'tamb' in name:
+            rename_dictionary[name] = 'Tamb'
+        elif 'wind' in name and 'speed' in name:
+            rename_dictionary[name] = 'wind_speed'
+        elif 'ghi' in name or ('global' in name and 'horizontal' in name):
+            rename_dictionary[name] = 'ghi'
+        elif 'dhi' in name or ('diffuse' in name and 'horizontal' in name):
+            rename_dictionary[name] = 'dhi'
+        elif 'datatime' in name or 'timestamp' in name:
+            rename_dictionary[name] = 'timestamp'
+
+    df.columns = columns
+    df.rename(columns = rename_dictionary, inplace=True)
+    df = df[rename_dictionary.values()]
+
+    return df
+
+class AnalysisPipeline(object):
+
+    def __init__(self, input_filename, system_metadata, clearsky = True):
+        '''
+        input_filename: csv with power time-series
+        system_metdata: dictionary with the following keys: 
+                        - systemid   (string. unique identifier) 
+                        - latitude
+                        - longitude
+                        - tilt       (degrees)
+                        - azimuth    (degrees)
+                        - pdc        (DC capacity in Watts)
+                        - tempco     (temperature coefficient)
```

Comment:
I would suggest `gamma_pmp` as the keyword

Reply:
Thank you, done!